spec/Spec/KZG.cry

```diff
@@ -108,6 +108,126 @@ verify_kzg_proof commitment_bytes z_bytes y_bytes proof_bytes =
+padding_zeros : [16][8] -> [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB][8]
+padding_zeros value = result where
+    result = (zero : [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB - 16][8]) # value
+
+/**
+ * Return the Fiat-Shamir challenge required by the rest of the protocol.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_challenge
+ * Discuss with Galois:
+ *      -   Overflow in data = blob + degree_poly + FIAT_SHAMIR_PROTOCOL_DOMAIN_with_leading_zeros + commitment_in_bytes_with_leading_zeros
+ */
+compute_challenge: Blob -> KZGCommitment -> BlsFieldElement
+compute_challenge blob commitment = result where
+    degree_poly = groupBy`{8,BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB} `(FIELD_ELEMENTS_PER_BLOB)
+    FIAT_SHAMIR_PROTOCOL_DOMAIN_with_leading_zeros = (zero : [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB - 16][8]) # FIAT_SHAMIR_PROTOCOL_DOMAIN
+    commitment_in_bytes = groupBy`{8, 48} commitment
+    commitment_in_bytes_with_leading_zeros = (zero : [BYTES_PER_FIELD_ELEMENT * FIELD_ELEMENTS_PER_BLOB - 48][8]) # commitment_in_bytes
+    data = blob + degree_poly + FIAT_SHAMIR_PROTOCOL_DOMAIN_with_leading_zeros + commitment_in_bytes_with_leading_zeros
+    result = hash_to_bls_field(data)
+
+/**
+ * Given a blob, return the KZG proof that is used to verify it against the commitment.
+ * This method does not verify that the commitment is correct with respect to `blob`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#compute_blob_kzg_proof
+ */
+compute_blob_kzg_proof: Blob -> Bytes48 -> KZGProof
+compute_blob_kzg_proof blob commitment_bytes = 
+    assert  ((length blob) == `(BYTES_PER_BLOB)) 
+            error_msg_for_blob 
+            assert  ((length commitment_bytes) == `(BYTES_PER_COMMITMENT))
+                    error_msg_for_commitment_bytes
+                    result where
+                        error_msg_for_blob = "The length of blob must be BYTES_PER_BLOB"
+                        error_msg_for_commitment_bytes = "The length of commitment_bytes be BYTES_PER_COMMITMENT"
+                        commitment = bytes_to_kzg_commitment commitment_bytes
+                        polynomial = blob_to_polynomial blob
+                        evaluation_challenge = compute_challenge blob commitment
+                        (proof, e) = compute_kzg_proof_impl polynomial evaluation_challenge
+                        result = proof
+
+/**
+ * Given a blob, return the KZG proof that is used to verify it against the commitment.
+ * This method does not verify that the commitment is correct with respect to `blob`.
+ * @see https://github.com/ethereum/consensus-specs/blob/dev/specs/deneb/polynomial-commitments.md#verify_blob_kzg_proof
+ */
+verify_blob_kzg_proof: Blob -> Bytes48 -> Bytes48 -> Bit
+verify_blob_kzg_proof blob commitment_bytes proof_bytes =
+    assert  ((length blob) == `(BYTES_PER_BLOB)) 
```

Comment:
This `assert` is always true because Cryptol is statically typed, so this code will not compile when the assert is false.